server/src/main/java/io/crate/planner/operators/WindowAgg.java

```diff
@@ -185,9 +185,17 @@ public ExecutionPlan build(DependencyCarrier executor,
                 sourcePlan.addProjection(projection);
             }
         } else {
-            sourcePlan.setDistributionInfo(new DistributionInfo(
-                DistributionType.MODULO,
-                source.outputs().indexOf(windowDefinition.partitions().iterator().next()))
+            Symbol firstPartition = windowDefinition.partitions().getFirst();
+            int index = source.outputs().indexOf(firstPartition);
+            // Approach 1
+            if (index == -1) {
+                //  We can have a top-level object column in outputs and partition by a sub-column.
+                //  Find index of the top column.
+                //  TODO, find a top level column of the firstPartition (is it always a ScopedSymbol?) and lookup by columnIdent in the outputs
+            }
```

Comment:
If I understand correctly, downside of this approach is that we set the whole object column as a distributeBy, ie it means that we would stream a potentially big object only to pick up one sub-column on other nodes?